stdlib/InteractiveUtils/src/codeview.jl

```diff
@@ -353,7 +353,7 @@ const llvm_types =
 const llvm_cond = r"^(?:[ou]?eq|[ou]?ne|[uso][gl][te]|ord|uno)$" # true|false
 
 function print_llvm_tokens(io, tokens)
-    m = match(r"^((?:[^\s:]+:)?)(\s*)(.*)", tokens)
+    m = match(r"^((?:[^(%\")\s:]+:)?)(\s*)(.*)", tokens)
```

Comment:
```suggestion
    m = match(r"^((?:[^\s%@][^\s:]*:)?)(\s*)(.*)", tokens)
```
⚠**MY SUGGESTION IS NOT TESTED**

I have my doubts if your regex is what you intended.
I think it is clearer in intent that it is not a variable, i.e., it does not start with `%` and `@`.

Reply:
Indeed, this didn't match the sequence `%"` but rather either one of `(%")`. I updated with matching the `%` at the beginning and added a case of string labels which is valid.

Reply:
I am not sure what your intention is with your change.
The regex of this line detects the label itself, not a reference to a label.

Reply:
Yes, what I am referring to is either one of:

```llvm
bla-bla: ; pred = %"another%block"
  br label %"another%block"
"another%block": ; pred = %bla-bla
  br label %bla-bla
```

![image](https://github.com/JuliaLang/julia/assets/9824244/4ef3a830-64c1-42cf-ba63-6749fddb4598)

Sorry, I meant matching `%` as actually *not* matching it (using `[^%]`), in the end I just removed `%` from the set of allowed character for the first case (`bla-bla` style).

Reply:
Instead of using an banlist, we could also use the following pattern:

https://github.com/llvm/llvm-project/blob/52271a5c11f6abde1fa1221db304212b5eb8ec7c/llvm/lib/AsmParser/LLLexer.cpp#L140

Reply:
We do not need a formal parser or lexer, since we can assume that the output of `code_llvm` is syntactically correct.
There are no formal assurances, but we can easily fix the problem, as we can do with this PR.